src/test/java/org/sagebionetworks/bridge/services/UploadServiceTest.java

```diff
@@ -1610,11 +1644,23 @@ private void mockTimelineMetadataForExporter3Test() {
         when(mockSchedule2Service.getTimelineMetadata(INSTANCE_GUID)).thenReturn(Optional.of(timelineMetadata));
     }
 
-    private List<AdherenceRecord> mockAdherenceForExporter3Test() {
-        // We don't actually need any of the data in the adherence record, just the fact that it exists.
-        List<AdherenceRecord> adherenceRecordList = ImmutableList.of(new AdherenceRecord());
-        PagedResourceList<AdherenceRecord> pagedResourceList = new PagedResourceList<>(adherenceRecordList, 1);
-        when(mockAdherenceService.getAdherenceRecords(eq(TEST_APP_ID), any())).thenReturn(pagedResourceList);
-        return adherenceRecordList;
+    private void mockAdherenceForExporter3Test() {
+        // We don't actually need any of the data in the adherence record. Just tag the adherence record with
+        // instanceGuid or uploadId, depending on what's passed in the AdherenceRecordSearch.
+        when(mockAdherenceService.getAdherenceRecords(eq(TEST_APP_ID), any())).thenAnswer(invocation -> {
+            AdherenceRecordsSearch search = invocation.getArgument(1);
```

Comment:
This is really handy. I didn't realize you could adjust what the mock returned based on the contents of what you passed in.